src/cpp/include/openvino/genai/llm_pipeline.hpp

```diff
@@ -0,0 +1,216 @@
+    
+    template <typename... Properties>
+    util::EnableIfAllStringAny<std::string, Properties...> generate(
+            std::string text,
+            Properties&&... properties) {
+        return generate(text, AnyMap{std::forward<Properties>(properties)...});
+    }
+    std::string generate(std::string text, const ov::AnyMap& config);
+
+    template <typename... Properties>
+    util::EnableIfAllStringAny<EncodedResults, Properties...> generate(
+        ov::Tensor input_ids,
+        Properties&&... properties) {
+        return generate(input_ids, AnyMap{std::forward<Properties>(properties)...});
+    }
+    EncodedResults generate(ov::Tensor input_ids, const ov::AnyMap& config);
+
+    /**
+    * @brief High level generate for batched prompts which encodes inputs and returns decoded outputs. 
+    * Streamer cannot be used for multibatch inputs.
+    *
+    * @param text input prompt
+    * @param generation_config optional GenerationConfig
+    * @return DecodedResults a structure with resulting texts & scores
+    */
+    DecodedResults generate(const std::vector<std::string>& texts, OptionalGenerationConfig generation_config);
+
+    /**
+    * @brief Low level generate to be called with already encoded input_ids tokens.
+    * Streamer cannot be used for multibatch inputs.
+    *
+    * @param input_ids encoded input prompt tokens
+    * @param attention_mask optional attention_mask
+    * @param generation_config optional GenerationConfig
+    * @param streamer optional streamer
+    * @return EncodedResults a structure with resulting tokens and scores
+    * @throws Exception if the stremaer is set for inputs_ids with multiple batches
+    */
+    EncodedResults generate(ov::Tensor input_ids, 
+                            std::optional<ov::Tensor> attention_mask, 
+                            OptionalGenerationConfig generation_config=std::nullopt,
+                            OptionalStreamerVariant streamer=std::nullopt);
+    
+    template <typename InputsType, typename... Properties>
+    util::EnableIfAllStringAny<std::string, Properties...> operator()(
+        InputsType text,
+        Properties&&... properties) {
+        return generate(text, AnyMap{std::forward<Properties>(properties)...});
+    }
+    
+    DecodedResults operator()(const std::vector<std::string>& text, OptionalGenerationConfig generation_config=std::nullopt) {
+        return generate(text, generation_config);
+    }
+
+    std::string operator()(
+        std::string text, 
+        OptionalGenerationConfig generation_config=std::nullopt, 
+        OptionalStreamerVariant streamer=std::nullopt
+    ) {
+        return generate(text, generation_config, streamer);
+    }
+    
+    ov::genai::Tokenizer get_tokenizer();
+    GenerationConfig get_generation_config() const;
+    void set_generation_config(const GenerationConfig& generation_config);
+
+    void start_chat();
+    void finish_chat();
+    void reset_state();
```

Comment:
`start_chat()` should  do what `reset_state()` does. That removes `reset_state()` from the API.

Reply:
done